### DISCUSSION TITLE
Can't find python

### DISCUSSION THREAD

**Discussion by ExissNA** [Panchovix/stable-diffusion-webui-reForge#130] [Q&A]
Didn't want to post a bug report because it's not a bug, it's user stupidity :) (I'm heavily hardware minded, not software)

I keep all of my installs separate because it minimizes chances that things will break each other.  I usually always do standalone installs.  So using GIT to install reForge, it's looking for python and can't find it.  Copied the folder for python from Forge standalone into the root directory, but I'm not sure exactly how to tell reforge "python is here."  Did I completely miss a version of reForge that has standalone python included?  

My thought was to run a version of reForge without a bunch of extensions (I have over 30K individual resources between LoRas and checkpoints) so the civit extensions wouldn't cause everything to absolutely die when trying to download/update preview/hash/whatever else.  I also wanted to try this out because I've seen numerous people recommending it, and while A1111 is getting better, it's still molasses in consideration.  Enabling tiled VAE helps a ton but it's still over a minute slower for a standard 1280x1440 ~30 steps with Forge and a 10GB 3080.  

Any help would be appreciated.

**Comment by AumGupta** [comment]
Hey @ExissNA 
You're on the right track with your approach to keep installs separate. This can indeed help prevent conflicts.

You can try the following to point reForge to your Python Installation:

**1. Add Python to PATH:**

* Go to your system settings or control panel.
* Search for "environment variables."
* Find the "Path" variable under "System variables."
* Click "Edit" and add the path to your Python installation (e.g., "C:\Python311").
* Save changes and restart your computer.

**2. Check reForge config:**

* Look for a file named `config.toml` or similar in your reForge installation directory.
* Search for a setting like `python_path`.
* If it exists, set it to the path of your Python installation.

**3. Create a virtual environment:**

* Open a terminal or command prompt.
* Navigate to your reForge project directory.
* Run the command: `python -m venv venv` (replace "venv" with your desired environment name).
* Activate the environment:
    * Windows: `venv\Scripts\activate`
    * macOS/Linux: `source venv/bin/activate`
* Install reForge and its dependencies using pip: `pip install reforge`

> Hope it helped.

**Reply by ExissNA** [reply]
Thank you for the response.  

#2 would be my desired choice, but there doesn't appear to be any kind of config.  If there is a config that is supposed to be generated after running, I don't quite get that far because it's a dead stop at "can't find python."  I may have to add it to path, though I would like to not do that if I can help it because I have 5 different web UI programs that may try and fight each other over the same python install.  

The only *.toml file that I found was "_typos.toml" and "pyproject.toml" - at least in the root directory.  My install folder is exactly as the current main branch is in content and structure, minus the other python I copied into the root.

**Reply by AumGupta** [reply]
I see. You can try one of the below:

1. Create separate virtual environments for each web UI program. This will isolate their Python installations and dependencies, preventing conflicts.
Use tools like `venv` or `virtualenv` to manage virtual environments.
Activate the appropriate virtual environment before running a program to ensure it uses the correct Python installation.

2. If you cannot use virtual environments, you can explicitly specify the path to the Python interpreter in the program's configuration or command-line arguments. This will ensure that the correct Python installation is used without modifying the system `PATH`.

3. If Python is installed in a non-standard location, you may need to set the `PYTHONPATH` environment variable to point to the directory containing the Python executable.

**Reply by ExissNA** [reply]
PaulZ0 mentioned "set PYTHON=path to python" and that's really what I was looking for, but forgot all of that was in the user-webui.bat file, but after doing that, it can't find or use "venv" so what I'm doing now is I deleted all of the contents of the root folder, am copying a pre-existing version of A1111, and will follow the directions listed for that particular install, merging, etc.  If that doesn't work, then I will follow the directions about the path.  If then, that doesn't work either, I will wait for the standalone version in which the "pre-done package is WIP."  It seems my install in its current state is missing a few things, I'm hoping this route will help rectify that.

**Comment by PaulZ0** [comment]
You can set your Python path in webui-user.bat

set PYTHON=path to python exe

**Reply by ExissNA** [reply]
This is great, thank you.  Now it's complaining about "unable to create venv."  

I missed the part in the original post about "_If you know what you are doing_, you can..."

**Reply by ExissNA** [reply]
I'm currently going to follow the "A1111 is already installed" method, I'm just copying things again since my A1111 is actually usable right now. It appears I am missing a few things still, this route may resolve it.
